week-1/Homework/mandatory/5-magic-8-ball.js

```diff
@@ -45,22 +45,58 @@ Very doubtful.
 
 // This should log "The ball has shaken!"
 // and return the answer.
-function shakeBall() {}
+
+function shakeBall() {
+  let possibleAnswer = [
+    "It is certain",
+    "It is decidedly so",
+    "Without a doubt",
+    "Yes - definitely",
+    "You may rely on it",
+    "As I see it, yes",
+    "Most likely",
+    "Outlook good",
+    "Yes",
+    "Signs point to yes",
+    "Reply hazy, try again",
+    "Ask again later",
+    "Better not tell you now",
+    "Cannot predict now",
+    "Concentrate and ask again","Don't count on it","My reply is no",
+  "My sources say no","Outlook not so good","Very doubtful"];
+
+  let answer = possibleAnswer[Math.floor(Math.random() * possibleAnswer.length)];
+  console.log(`The ball has shaken!`)
+  return answer;
+}
 
 // The answer should come from shaking the ball
-let answer;
+let answer = shakeBall();
+console.log(answer);
 
 // When checking the answer, we should tell someone if the answer is
 // - very positive
 // - positive
 // - negative
 // - very negative
-function checkAnswer() {}
+function checkAnswer(answer) {
+  let getIndexNum = possibleAnswer.indexOf(answer);
```

Comment:
This line doesn't quite work, because `possibleAnswer` is a local variable which only exists inside the `shakeBall` function, so you can't see it here. If you move `possibleAnswer` outside of `shakeBall`, so that it exists at the top-level of this file, this should work fine :)